benchmark/utils/main.swift

```diff
@@ -400,4 +410,12 @@ registerBenchmark(Walsh)
 registerBenchmark(WordCount)
 registerBenchmark(XorLoop)
 
-main()
+if #available(macOS 9999, *) {
+  
+  Task.runDetached {
+    await main()
+    exit(0)
+  }
+  
+  sleep(100000)
```

Comment:
If you have control over how this gets compiled and can pass `parse-as-library` to the invocation, wrapping the detached run in
```
@main struct Main {
    static func main() async {
      // The code
     }
}
```

will give you an async context that you can await on instead of sleeping for a kajillion moments.

Reply:
As a temporary workaround you can use ``runAsyncAndBlock``